### PR TITLE
N: 5 new ICollection<T> extension methods (closes #48–52, bumps to v0.3.0)

### LINKED ISSUE #48: Add RemoveRange extension method
## Description
Add a `RemoveRange` extension method to remove multiple items from any `ICollection<T>` in a single call, mirroring the existing `AddRange` method.

## Proposed Signature
```csharp
public static void RemoveRange<T>(this ICollection<T> source, IEnumerable<T> items)
```

## Example Usage
```csharp
ICollection<string> names = new List<string> { "Alice", "Bob", "Charlie", "Dave" };
names.RemoveRange(new[] { "Bob", "Dave" });
// names: "Alice", "Charlie"
```

## Edge Cases to Consider
- Null source / null items → `ArgumentNullException`
- Read-only collection → `NotSupportedException`
- Items not in collection → silently skip (mirrors `ICollection<T>.Remove` behavior)
- Duplicate items in the removal list → each call to `Remove` removes one occurrence
- Empty items enumerable → no-op
- Thread safety — not thread-safe, document accordingly
- Performance: O(n*m) where n = items to remove, m = collection size for list-based collections

## Return Value Consideration
Consider returning `int` (count of items actually removed) instead of `void`, similar to `HashSet<T>.RemoveWhere`.

### PR DESCRIPTION
**Pull request by Chris-Wolfgang**
## Summary

Closes #48, #49, #50, #51, #52.

Adds five new public methods (one of them with two overloads), bringing
the ICollection surface from 3 to 9 methods. This is a **MINOR API
addition** — the next release should be v0.3.0 rather than v0.2.1.

## New surface

| Method | Returns | Issue |
|---|---|---|
| `RemoveRange<T>(this ICollection<T>, IEnumerable<T>)` | `void` | #48 |
| `AddRangeIf<T>(this ICollection<T>, IEnumerable<T>, Func<T, bool>)` | `void` | #49 |
| `RemoveWhere<T>(this ICollection<T>, Func<T, bool>)` | `int` | #50 |
| `ReplaceAll<T>(this ICollection<T>, IEnumerable<T>)` | `void` | #51 |
| `AddIfNotContains<T>(this ICollection<T>, T)` | `bool` | #52 |
| `AddIfNotContains<T>(this ICollection<T>, IEnumerable<T>)` | `int` | #52 |

## Tests

27 new tests covering null guards, baseline behaviour, predicate
gating, count returns, and edge cases (empty input, removing every
item, deduplication within bulk add). All 66 tests pass on net10.0
locally; build clean (0 warnings, 0 errors) with TWE active.

## Docs updated in sync

- **README.md** — 7-row API table replaces the 3-row table.
- **CHANGELOG.md** — `[Unreleased]` populated with the new surface.
- **`.github/copilot-instructions.md`** — extension-method list
  updated.

## Stacking

Top of the vNext stack. Bases on `feature/114-benchmark-twe` (PR #129).

### CODE REVIEW

File: src/Wolfgang.Extensions.ICollection/ICollectionExtensions.cs [hunk 1]
```diff
@@ -227,4 +227,270 @@ public static bool IsNotEmpty<T>(this ICollection<T> source)
 
         return source.Count > 0;
     }
+
+
+
+    /// <summary>
+    /// Removes every occurrence of each item in <paramref name="items"/>
+    /// from <paramref name="source"/>.
+    /// </summary>
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply src/Wolfgang.Extensions.ICollection/ICollectionExtensions.cs:236]: Fixed in 4a54ea3 — XML summary now reads 'Removes one occurrence of each item in items' and explicitly notes that duplicates in items therefore remove additional occurrences (one per call to ICollection<T>.Remove).

File: src/Wolfgang.Extensions.ICollection/ICollectionExtensions.cs [hunk 2]
```diff
@@ -227,4 +227,270 @@ public static bool IsNotEmpty<T>(this ICollection<T> source)
...
+            throw new ArgumentNullException(nameof(source));
+        }
+
+        if (predicate is null)
+        {
+            throw new ArgumentNullException(nameof(predicate));
+        }
+
+        var toRemove = new List<T>();
+        foreach (var item in source)
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply src/Wolfgang.Extensions.ICollection/ICollectionExtensions.cs:369]: Fixed in 4a54ea3 — added a HashSet<T> fast path that delegates to HashSet<T>.RemoveWhere (native, no temp-list allocation). The general path is unchanged because mutation-during-enumeration is undefined for the general ICollection<T> case. List<T> isn't special-cased because List<T>.RemoveAll only landed in netstandard2.1 and the in-place pattern is fast enough for typical sizes.

File: tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs [hunk 3]
```diff
@@ -477,4 +477,306 @@ private static LinkedList<T> LinkedListWith<T>(params T[] items)
...
+            new System.Collections.ObjectModel.ReadOnlyCollection<int>(new List<int> { 1, 2 });
+        Assert.Throws<NotSupportedException>(() => source.RemoveRange(new[] { 1 }));
+    }
+
+
+
+    // =========================================================================
+    // AddRangeIf tests
+    // =========================================================================
+
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.Extensions.ICollection.Tests.Unit/ICollectionExtensionTests.cs:559]: Fixed in 4a54ea3 — added 5 ReadOnly NotSupportedException tests covering AddRangeIf, RemoveWhere, ReplaceAll, AddIfNotContains(T), and AddIfNotContains(IEnumerable<T>). 71 tests total on net10.0, build clean.
